### PR TITLE
Bluetooth: Controller: Fix Extended Scan and Periodic Sync terminate

### PR DESCRIPTION
**Pull request by cvinayak**
Fix Extended Scan disable and Periodic Synchronization
terminate to cover scenarios where chain PDU reception could
complete while thread context is waiting for the radio event
to be done.

Changes here ensure auxiliary context is not released twice
which is caught as an assertion in flush() when checking for
validity of aux->parent pointer.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/ull_scan_aux.c [hunk 1]
```diff
@@ -725,24 +719,28 @@ void ull_scan_aux_setup(memq_link_t *link, struct node_rx_hdr *rx)
...
-				aux->rx_last->rx_ftr.extra = rx;
-				aux->rx_last = rx;
+			if (unlikely(scan->is_stop)) {
+				return;
 			}
+
+			aux->rx_last->rx_ftr.extra = rx;
+			aux->rx_last = rx;
 		} else {
+			struct ll_sync_set *sync;
```
Comment (Thalley) [inline subsys/bluetooth/controller/ll_sw/ull_scan_aux.c:729]:
```suggestion
			const struct ll_sync_set *sync;
```

File: subsys/bluetooth/controller/ll_sw/ull_scan_aux.c [hunk 2]
```diff
@@ -1059,6 +1060,11 @@ static inline struct ll_sync_iso_set *
 
 static void done_disabled_cb(void *param)
 {
+	struct ll_scan_aux_set *aux;
+
+	aux = param;
+	LL_ASSERT(aux->parent);
+
 	flush(param);
```
Comment (Thalley) [inline subsys/bluetooth/controller/ll_sw/ull_scan_aux.c:1068]: Would it make sense to do `flush(aux)` now that we have a more specific name? (not needed at all). 

If not, I guess we could also add `const` to `struct ll_scan_aux_set *aux;` or even shorten this to 
`LL_ASSERT(((struct ll_scan_aux_set *)param)->parent)`

**Review by Thalley — DISMISSED** [review]
Minor comments, and is fine to go in without changes

**Review by Thalley — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]
